File: 27.Remove_Element/main.java
import java.util.Arrays;

public class Main {
    public static void main(String[] args) {
        int[] nums = {0, 1, 2, 2, 3, 0, 4, 2};
        int val = 2;
        Solution solution = new Solution();
        int k = solution.removeElement(nums, val);

        System.out.println("k = " + k);
        System.out.println("Modified nums = " + Arrays.toString(Arrays.copyOf(nums, k)));
    }
}

class Solution {
    public int removeElement(int[] nums, int val) {
        int k = 0;
        for (int num : nums) {
            if (num != val) {
                nums[k++] = num;
            }
        }
        return k;
    }
}

File: 27.Remove_Element/main.py
def main():
    nums = [0, 1, 2, 2, 3, 0, 4, 2]
    val = 2
    solution = Solution()
    k = solution.removeElement(nums, val)
    print("k =", k)
    print("Modified nums =", nums[:k])

class Solution:
    def removeElement(self, nums, val):
        k = 0
        for num in nums:
            if num != val:
                nums[k] = num
                k += 1
        return k

if __name__ == "__main__":
    main()
